src/memory_manager.rs

```diff
@@ -510,6 +501,20 @@ pub fn initialize_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>, tls: VMThre
     probe!(mmtk, collection_initialized);
 }
 
+/// Ask GC worker threads to exit.  Currently, this function is for the sole purpose of forking.
```

Comment:
I think it would be prudent to both mention in the API docs and test (via the MockVM) what happens in cases when:

1. A GC is required, but all GC threads have exited
2. A GC is in progress, and `pre_fork` is called

Reply:
I changed the comments to make it clear that 

1.  The VM must not attempt to allocate between `prepare_to_fork` and `after_fork`.
2.  Doing GC and preparing for forking are mutually exclusive, and GC workers must finish one of them before starting the other.